tools/speech_data_explorer/data_explorer.py

```diff
@@ -1,846 +1,1044 @@
-
-    data = []
-    wer_dist = 0.0
-    wer_count = 0
-    cer_dist = 0.0
-    cer_count = 0
-    wmr_count = 0
-    wer = 0
-    cer = 0
-    wmr = 0
-    mwa = 0
-    num_hours = 0
-    vocabulary = defaultdict(lambda: 0)
-    alphabet = set()
-    match_vocab = defaultdict(lambda: 0)
-
-    sm = difflib.SequenceMatcher()
-    metrics_available = False
-    with open(data_filename, 'r', encoding='utf8') as f:
-        for line in tqdm.tqdm(f):
-            item = json.loads(line)
-            if not isinstance(item['text'], str):
-                item['text'] = ''
-            num_chars = len(item['text'])
-            orig = item['text'].split()
-            num_words = len(orig)
-            for word in orig:
-                vocabulary[word] += 1
-            for char in item['text']:
-                alphabet.add(char)
-            num_hours += item['duration']
-
-            if 'pred_text' in item:
-                metrics_available = True
-                pred = item['pred_text'].split()
-                measures = jiwer.compute_measures(item['text'], item['pred_text'])
-                word_dist = measures['substitutions'] + measures['insertions'] + measures['deletions']
-                char_dist = editdistance.eval(item['text'], item['pred_text'])
-                wer_dist += word_dist
-                cer_dist += char_dist
-                wer_count += num_words
-                cer_count += num_chars
-
-                sm.set_seqs(orig, pred)
-                for m in sm.get_matching_blocks():
-                    for word_idx in range(m[0], m[0] + m[2]):
-                        match_vocab[orig[word_idx]] += 1
-                wmr_count += measures['hits']
-
-            data.append(
-                {
-                    'audio_filepath': item['audio_filepath'],
-                    'duration': round(item['duration'], 2),
-                    'num_words': num_words,
-                    'num_chars': num_chars,
-                    'word_rate': round(num_words / item['duration'], 2),
-                    'char_rate': round(num_chars / item['duration'], 2),
-                    'text': item['text'],
-                }
-            )
-            if metrics_available:
-                data[-1]['pred_text'] = item['pred_text']
-                if num_words == 0:
-                    num_words = 1e-9
-                if num_chars == 0:
-                    num_chars = 1e-9
-                data[-1]['WER'] = round(word_dist / num_words * 100.0, 2)
-                data[-1]['CER'] = round(char_dist / num_chars * 100.0, 2)
-                data[-1]['WMR'] = round(measures['hits'] / num_words * 100.0, 2)
-                data[-1]['I'] = measures['insertions']
-                data[-1]['D'] = measures['deletions']
-                data[-1]['D-I'] = measures['deletions'] - measures['insertions']
-
-            if estimate_audio:
-                filepath = absolute_audio_filepath(item['audio_filepath'], data_filename)
```

Comment:
Please integrate Ante's updates back (to support audio_base_path)

Reply:
done